src/main/java/de/wirvsvirus/hack/service/StatsCalculationLogic.java

```diff
@@ -0,0 +1,111 @@
+package de.wirvsvirus.hack.service;
+
+import static de.wirvsvirus.hack.service.IntervalIntersectionUtil.INSTANT_MAX;
+import static de.wirvsvirus.hack.service.IntervalIntersectionUtil.INSTANT_MIN;
+import static de.wirvsvirus.hack.service.IntervalIntersectionUtil.intersectList;
+import static de.wirvsvirus.hack.service.IntervalIntersectionUtil.truncateListUnit;
+import de.wirvsvirus.hack.model.Sentiment;
+import de.wirvsvirus.hack.model.UserGroupMembershipHistory;
+import de.wirvsvirus.hack.model.UserGroupMembershipHistory.Change;
+import de.wirvsvirus.hack.model.UserStatusChangeHistory;
+import de.wirvsvirus.hack.service.StatsService.MembershipPoints;
+import de.wirvsvirus.hack.service.StatsService.StatusPoints;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import one.util.streamex.MoreCollectors;
+import one.util.streamex.StreamEx;
+import org.apache.commons.lang3.tuple.Pair;
+
+public final class StatsCalculationLogic {
+
+
+  public static Duration calcSunshineHoursInGroup(
+      final List<UserStatusChangeHistory> historyStatusChanges,
+      final List<UserGroupMembershipHistory> historyGroupMembership,
+      final UUID userId,
+      final UUID groupId,
+      final Instant now) {
+
+    final List<Pair<Instant, Instant>> membershipIntervals =
+        StreamEx.of(historyGroupMembership)
```

Comment:
Vielleicht noch ein Check, dass der erste Change ein JOIN oder START ist?

Reply:
das sollte in Zeile 40 mit geprüft werden